components/eam/src/physics/cam/zm_microphysics.F90

```diff
@@ -0,0 +1,3337 @@
+!---------------------------------------------------------------------------------
+
+use shr_kind_mod,      only: r8=>shr_kind_r8
+use spmd_utils,        only: masterproc    
+use ppgrid,            only: pcols, pver, pverp
+use physconst,         only: gravit, rair, tmelt, cpair, rh2o, r_universal, mwh2o, rhoh2o
+use physconst,         only: latvap, latice
+use activate_drop_mam, only: actdrop_mam_calc
+use ndrop_bam,         only: ndrop_bam_run
+use nucleate_ice_conv, only: nucleati_conv
+#ifndef HAVE_ERF_INTRINSICS
+use shr_spfn_mod,      only: erf => shr_spfn_erf
+#endif
+use shr_spfn_mod, only: gamma => shr_spfn_gamma
+use wv_saturation,  only: svp_water, svp_ice
+use cam_logfile,       only: iulog
+use cam_abortutils,    only: endrun
+implicit none
+private
+save
+
+public :: &
+   zm_mphyi, &
+   zm_mphy,  &
+   zm_aero_t,  &
+   zm_microp_st
+
+! Private module data
+
+! constants remaped
+real(r8) :: g      ! gravity
+real(r8) :: mw     ! molecular weight of water
+real(r8) :: r      ! Dry air Gas constant
+real(r8) :: rv     ! water vapor gas contstant
+real(r8) :: rr     ! universal gas constant
+real(r8) :: cpp    ! specific heat of dry air
+real(r8) :: rhow   ! density of liquid water
+real(r8) :: xlf    ! latent heat of freezing
+
+!from 'microconstants'
+real(r8) :: rhosn  ! bulk density snow
+real(r8) :: rhoi   ! bulk density ice
+real(r8) :: rhog   ! bulk density graupel
+
+real(r8) :: ac,bc,as,bs,ai,bi,ar,br,ag,bg  !fall speed parameters 
+real(r8) :: ci,di    !ice mass-diameter relation parameters
+real(r8) :: cs,ds    !snow mass-diameter relation parameters
+real(r8) :: cr,dr    !drop mass-diameter relation parameters
+real(r8) :: cg,dg    !graupel mass-diameter relation parameters
+real(r8) :: Eii      !collection efficiency aggregation of ice
+real(r8) :: Ecc      !collection efficiency
+real(r8) :: Ecr      !collection efficiency cloud droplets/rain
+real(r8) :: ecg      ! collection efficiency, ice-droplet collisions
+real(r8) :: DCS      !autoconversion size threshold
+real(r8) :: bimm,aimm !immersion freezing
+real(r8) :: rhosu     !typical 850mn air density
+real(r8) :: mi0       ! new crystal mass
+real(r8) :: mg0       ! mass of embryo graupel
+real(r8) :: rin       ! radius of contact nuclei
+real(r8) :: pi        ! pi
+real(r8) :: mmult
+
+! for Bergeron process (Rotstayn et al.2000)
+real(r8) :: Ka_b       ! thermal conductivity of air(J/m/s/K)
+real(r8) :: Ls_b       ! latent heat of sublimation of water(J/kg)
+real(r8) :: Rv_b       ! specigic gas constant for water vapour(J/kg/K)
+real(r8) :: alfa_b     ! parameter for ice crystal habit
+real(r8) :: rhoi13     ! rhoi**(1/3)
+real(r8) :: c23        ! 2/3  
+
+real(r8) ::  cons14,cons16,cons17,cons18,cons19,cons24,cons25, cons31, cons32, cons41
+
+real(r8) :: droplet_mass_25um
+
+! contact freezing due to dust
+! dust number mean radius (m), Zender et al JGR 2003 assuming number mode radius of 0.6 micron, sigma=2
```

Comment:
@yfenganl , could these hardwired parameters related to dust particles have potential conflict with the new dust emission scheme?

Reply:
@hwangacme @yfenganl I think these rn_dst are for bulk aerosol scheme not MAM4, which is not actually used.